lib/spack/spack/test/environment_modifications.py

```diff
@@ -70,9 +71,21 @@ def prepare_environment_for_tests(working_env):
     """
     os.environ["UNSET_ME"] = "foo"
     os.environ["EMPTY_PATH_LIST"] = ""
-    os.environ["PATH_LIST"] = "/path/second:/path/third"
-    os.environ["REMOVE_PATH_LIST"] = "/a/b:/duplicate:/a/c:/remove/this:/a/d:/duplicate/:/f/g"
-    os.environ["PATH_LIST_WITH_SYSTEM_PATHS"] = "/usr/include:" + os.environ["REMOVE_PATH_LIST"]
+    list_sep = ";" if is_windows else ":"
+    os.environ["PATH_LIST"] = list_sep.join([os.sep.join(["","path","second"]), os.sep.join(["","path", "third"])])
+    os.environ["REMOVE_PATH_LIST"] = list_sep.join([
+        os.sep.join(["","a", "b"]),
+        os.sep.join(["","duplicate"]),
+        os.sep.join(["","a", "c"]),
+        os.sep.join(["","remove", "this"]),
+        os.sep.join(["","a","d"]),
+        os.sep.join(["","duplicate"]),
+        os.sep.join(["","f","g"])
+        ])
```

Comment:
This feels like horrible syntax but at least it's the same code for both platforms? any recommendations?